archinfo/arch_soot.py

```diff
@@ -40,13 +45,56 @@ def from_string(cls, tstr):
     def fullname(self):
         return "%s.%s" % (self.class_name, self.name)
 
+    def matches_with_native_name(self, native_method):
+        """
+        Name of native methods are getting encoded, s.t. they translate into valid C function names.
+        :return: True, if name of soot method matches the mangled native name.
+        """
+        
+        if "__" in native_method:
+            # if native methods are overloaded, two underscores are used
+            # TODO check function signature
+            raise NotImplementedError('Overloaded native methods are not supported.')
+
+        # demangle native name
+        native_method =  native_method.replace('_1', '_')
+        # TODO unicode escaping
+
+        method_native_name = "Java_{class_name}_{method_name}".format(
+                              class_name=self.class_name,
+                              method_name=self.name)
+        return native_method == method_native_name
+
     @classmethod
-    def from_method(cls, method):
-        return cls(method.class_name, method.name, method.params)
+    def from_soot_method(cls, soot_method):
+        return cls(class_name=soot_method.class_name, 
+                   name=soot_method.name, 
+                   params=soot_method.params, 
+                   soot_method=soot_method)
```

Comment:
The right parenthesis should go to the next line.

```
    soot_method=soot_method,
    )
```

Reply:
I... disagree with this

Reply:
PEP8? Or whatever coding style I've been following.

Reply:
it's well known that following pep8 religiously is bad

like I see the benefit to doing it that way, you get to have the trailing comma like in dict/list literals, but it looks soooooooooo ugly to my eyes...

Reply:
_welp_

Fine. I don't really care. Nitpicks anyway.